types/module/module.go

```diff
@@ -370,9 +391,22 @@ func (m *Manager) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, genesisData
 			continue
 		}
 
-		if module, ok := m.Modules[moduleName].(HasGenesis); ok {
-			ctx.Logger().Debug("running initialization for module", "module", moduleName)
+		mod := m.Modules[moduleName]
 
+		ctx.Logger().Debug("running initialization for module", "module", moduleName)
+
+		if module, ok := mod.(appmodule.HasGenesis); ok {
+			// core API genesis
+			source, err := genesis.SourceFromRawJSON(genesisData[moduleName])
+			if err != nil {
+				panic(err)
+			}
+
+			err = module.InitGenesis(ctx, source)
+			if err != nil {
+				panic(err)
+			}
+		} else if module, ok := mod.(HasGenesis); ok {
```

Comment:
Why we needed these checks?

Reply:
they are two different interfaces but we still want to support the old ones